fe/fe-core/src/main/java/org/apache/doris/rewrite/RewriteFromUnixTimeRule.java

```diff
@@ -0,0 +1,107 @@
+
+import org.apache.doris.analysis.Analyzer;
+import org.apache.doris.analysis.BinaryPredicate;
+import org.apache.doris.analysis.BoolLiteral;
+import org.apache.doris.analysis.CompoundPredicate;
+import org.apache.doris.analysis.Expr;
+import org.apache.doris.analysis.FunctionCallExpr;
+import org.apache.doris.analysis.FunctionParams;
+import org.apache.doris.analysis.LiteralExpr;
+import org.apache.doris.analysis.SlotRef;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.common.AnalysisException;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+/*
+* rewrite sql: select * from table where from_unixtime(query_time) > '2021-03-02 12:12:23'
+* into: select * from table where query_time > 1614658343
+* query_time is integer type
+* 1614658343 is the timestamp of 2021-03-02 12:12:23
+* this rewrite can improve the query performance, because from_unixtime is cpu-exhausted
+* */
+public class RewriteFromUnixTimeRule implements ExprRewriteRule {
+    public static RewriteFromUnixTimeRule INSTANCE = new RewriteFromUnixTimeRule();
+    @Override
+    public Expr apply(Expr expr, Analyzer analyzer) throws AnalysisException {
+        if (!(expr instanceof BinaryPredicate)) {
+            return expr;
+        }
+        BinaryPredicate bp = (BinaryPredicate) expr;
+        Expr left = bp.getChild(0);
+        if (!(left instanceof FunctionCallExpr)) {
+            return expr;
+        }
+        FunctionCallExpr fce = (FunctionCallExpr) left;
+        if (!fce.getFnName().getFunction().equalsIgnoreCase("from_unixtime")) {
+            return expr;
+        }
+        FunctionParams params = fce.getParams();
+        if (params == null) {
+            return expr;
+        }
+        //definition: from_unixtime(int, format)
+        if (params.exprs().size() != 1 && params.exprs().size() != 2) {
+            return expr;
+        }
+        Expr paramSlot = params.exprs().get(0);
+        if (!(paramSlot instanceof SlotRef)) {
+            return expr;
+        }
+        SlotRef sr = (SlotRef) paramSlot;
+        if (!sr.getColumn().getType().isIntegerType()) {
+            return new BoolLiteral(false);
+        }
+        Expr right = bp.getChild(1);
+        if (!(right instanceof LiteralExpr)) {
+            return expr;
+        }
+        LiteralExpr le = (LiteralExpr) right;
+        SimpleDateFormat format =   new SimpleDateFormat( "yyyy-MM-dd" );
+        //default format is "yyyy-MM-dd HH:mm:ss"
+        if (params.exprs().size() == 1) {
+            format = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss" );
+        } else {
+            LiteralExpr fm = (LiteralExpr) params.exprs().get(1);
+            format = new SimpleDateFormat(fm.getStringValue());
+        }
+        try {
+            Date date = format.parse(le.getStringValue());
+            //it is better to add low bound 0, because when a field contains negative data like -100, it will be queried as a result
```

Comment:
Why do we need to add low bound or upper bound explicitly?

Reply:
rewrite sql: select * from table where from_unixtime(query_time) < '2021-03-02 12:12:23'
 into: select * from table where query_time < 1614658343

if query_time contains a data like -100, query: select * from table where query_time < 1614658343 will output -100, that is incorrect.

the timestamp range is [ 0,  253402271999],  1970-01-01 00:00:00 ~ 9999-12-31 23:59:59

Reply:
OK, I see.  The comment "it will be queried as a result" maybe need to change.  "it is better " should change to must.

Reply:
ok